gix-blame/src/file/function.rs

```diff
@@ -144,92 +141,42 @@ where
                 &mut buf3,
             )?;
             let Some(modification) = changes_for_file_path else {
-                // None of the changes affected the file we’re currently blaming. Pass blame to parent.
-                for unblamed_hunk in &mut hunks_to_blame {
-                    unblamed_hunk.pass_blame(suspect, parent_id);
-                }
-                continue;
-            };
-
-            match modification {
-                gix_diff::tree::recorder::Change::Addition { .. } => {
-                    // Every line that has not been blamed yet on a commit, is expected to have been
-                    // added when the file was added to the repository.
-                    out.extend(
-                        hunks_to_blame
-                            .iter()
-                            .map(|hunk| BlameEntry::from_unblamed_hunk(hunk, suspect)),
-                    );
-
-                    hunks_to_blame.clear();
-                    break;
-                }
-                gix_diff::tree::recorder::Change::Deletion { .. } => todo!(),
-                gix_diff::tree::recorder::Change::Modification { previous_oid, oid, .. } => {
-                    let changes = blob_changes(&odb, resource_cache, oid, previous_oid, file_path, &mut stats)?;
-                    hunks_to_blame = process_changes(&mut out, hunks_to_blame, changes, suspect);
-                    for unblamed_hunk in &mut hunks_to_blame {
-                        unblamed_hunk.pass_blame(suspect, parent_id);
-                    }
-                }
-            }
-        } else {
-            for parent_id in &parent_ids {
-                if let Some(parent_entry) =
-                    find_path_entry_in_commit(&odb, parent_id, file_path, &mut buf, &mut buf2, &mut stats)?
-                {
-                    if entry.oid == parent_entry.oid {
-                        // The blobs storing the blamed file in `entry` and `parent_entry` are
-                        // identical which is why we can pass blame to the parent without further
-                        // checks.
-                        for unblamed_hunk in &mut hunks_to_blame {
-                            unblamed_hunk.pass_blame(suspect, *parent_id);
-                        }
-                        continue 'outer;
-                    }
-                }
-            }
```

Comment:
Did you benchmark this removal? From what I can remember this is a shortcut (that I unfortunately did not add a comment to :-)). We go through `parent_ids` twice which can seem wasteful at first, but since the `entry.oid == parent_entry.oid` shortcut will allow us to completely skip diffing to any parent, it amortizes quickly as far as I remember. Otherwise, we might diff to a previous parent only to find that we can completely pass blame to a subsequent one. (Also, if this shortcut is still present somewhere, feel free to ignore this comment.)

Reply:
Thanks for elaborating! 
It's good to know the code is still present, and now looks like this:

```rust
        for parent_id in &parent_ids {
            if let Some(parent_entry) =
                find_path_entry_in_commit(&odb, parent_id, file_path, &mut buf, &mut buf2, &mut stats)?
            {
                if entry.oid == parent_entry.oid {
                    pass_blame_from_to(suspect, *parent_id, &mut hunks_to_blame);
                    continue 'outer;
                }
            }
        }
```

Reply:
Great, then I just missed that part, thanks for the hint!